### PR TITLE
[WIP] docs: Add `gci-git-guide.md`.

### PR DESCRIPTION
**Pull request by HarshitOnGitHub**
This adds a shortened version of Zulip's git guide aimed at presenting to GCI students only what they require while starting. The original git guide is too long and students are likely to skip it. This is still kind of way longer than ideal @derAnfaenger @lonerz @sampritipanda @tommyip, can anyone of you give this a read and suggest what else could be added/deleted?

P.S.: I haven't yet cleaned up the doc completely, just put up here for initial feedback.

### CODE REVIEW

File: docs/gci-git-guide.md [hunk 1]
```diff
@@ -0,0 +1,1041 @@
+# GCI Git & GitHub Guide
+
+This a shortened version of zulip's git guide specifically focused on GCI
```
Comment (lonerz) [inline docs/gci-git-guide.md]: r/zulip/Zulip

File: docs/gci-git-guide.md [hunk 2]
```diff
@@ -0,0 +1,1041 @@
...
+  [linters][zulip-rtd-lint-tools], [tests][zulip-rtd-testing], continuous
+  integration with [Travis CI][travis-ci], and [mypy][zulip-rtd-mypy].
+
+- We use [zulipbot][zulip-rtd-zulipbot-usage] to manage our issues and
+pull requests to create a better GitHub workflow for contributors.
+
+***
+
+The following sections will help you be awesome with Zulip and Git/GitHub in a
+rebased-based workflow. Read through it if you're new to git, to a rebase-based
```
Comment (lonerz) [inline docs/gci-git-guide.md]: r/git/Git

File: docs/gci-git-guide.md [hunk 3]
```diff
@@ -0,0 +1,1041 @@
...
+git workflow, or if you'd like a git refresher.
```
Comment (lonerz) [inline docs/gci-git-guide.md]: r/git/Git

File: docs/gci-git-guide.md [hunk 4]
```diff
@@ -0,0 +1,1041 @@
...
+- [Configure Git][gitbook-config] with your name and email.  We
+  recommend using your full name (not just your first name), since
+  that's what we'll use to give credit to your work in places like the
+  Zulip release notes.
+- Install the command auto-completion and/or git-prompt plugins available for
+  [Bash][gitbook-other-envs-bash] and [Zsh][gitbook-other-envs-zsh].
+
+### Get a graphical client
+
+Even if you're comfortable using git on the command line, having a graphic
```
Comment (lonerz) [inline docs/gci-git-guide.md]: r/git/Git

File: docs/gci-git-guide.md [hunk 5]
```diff
@@ -0,0 +1,1041 @@
+# GCI Git & GitHub Guide
+
+This a shortened version of zulip's git guide specifically focused on GCI
+students. You can read the original version [here][zulip-git-guide].
```
Comment (derAnfaenger) [inline docs/gci-git-guide.md]: Is this supposed to be broken, so we can insert the appropriate link once it's in a good location?

File: docs/gci-git-guide.md (hunk 1 above)
Comment (derAnfaenger) [reply docs/gci-git-guide.md]: r/git/Git

**Comment by timabbott** [comment]
Yeah, I think splitting the git-guide is a good idea.  We might want to do this in the main repo, but holding off a couple days for https://github.com/zulip/zulip/pull/7326 to get merged, since I think that will make it not cost as much screen real estate to have several sub-articles...

**Comment by sampritipanda** [comment]
The setting up and clone part will probably be covered by some of the beginner tasks like making a pull request, and setting up developer environment.

 I think we should instead have something like a git cheatsheet for gci where you have a list of commands for common tasks like rebase/amend/squash.

**Comment by derAnfaenger** [comment]
@sampritipanda we have https://github.com/zulip/zulip-gci/blob/master/docs/git-cheat-sheet.md and https://github.com/zulip/zulip-gci/blob/master/docs/git-cheat-detailed.md

**Comment by timabbott** [comment]
Strategically, what I think we might want to do is work on splitting `git-guide.md` in the main Zulip project so everyone can benefit.  Now that we have the new ReadTheDocs sidebar structure, I think we could make a "Git guide" collapsable section as the path forward for this.  What do you think @HarshitOnGitHub?  

We can have the URLs  be e.g.: zulip.readthedocs.io/en/latest/git/* (aka put them in a docs/git/ directory)

What I'd start with is just cutting it up into ~10 articles like this (basically 1 per section):
----------------------
overview.md: Quick start: How Zulip uses Git and GitHub
----------------------
setup.md: Set up Git
----------------------
the-git-difference.md: How Git is different
----------------------
terminology.md: Important Git terms
----------------------
cloning.md: Get Zulip code
----------------------
using.md: Using Git as you work
----------------------
pull-requests.md: 
Create a pull request
Update a pull request
----------------------
collaborating.md: Collaborate
----------------------
reviewing.md: Review changes
----------------------
troubleshooting.md: Get and stay out of trouble
----------------------
zulip-tools.md: Zulip-specific tools
----------------------

If we're going to do that, I think the right approach is to first do a PR that just splits the files out, and doesn't make any other changes beyond updating previously doc-internal links.  And then we can do a series of smaller commits to make it more readable.

**Comment by HarshitOnGitHub** [comment]
Opened https://github.com/zulip/zulip/pull/7509 for splitting git-guide.md, so I think this is no longer required.

**Pull request closed without merging**